src/plugins/template/src/async_infer_request.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "async_infer_request.hpp"
+
+#include "infer_request.hpp"
+#include "openvino/runtime/iinfer_request.hpp"
+#include "template_itt.hpp"
+
+// ! [async_infer_request:ctor]
+TemplatePlugin::AsyncInferRequest::AsyncInferRequest(const std::shared_ptr<TemplatePlugin::InferRequest>& inferRequest,
+                                                     const InferenceEngine::ITaskExecutor::Ptr& cpuTaskExecutor,
+                                                     const InferenceEngine::ITaskExecutor::Ptr& waitExecutor,
+                                                     const InferenceEngine::ITaskExecutor::Ptr& callbackExecutor)
+    : ov::IAsyncInferRequest(inferRequest, cpuTaskExecutor, callbackExecutor),
+      m_wait_executor(waitExecutor) {
+    // In current implementation we have CPU only tasks and no needs in 2 executors
+    // So, by default single stage pipeline is created.
+    // This stage executes InferRequest::infer() using cpuTaskExecutor.
+    // But if remote asynchronous device is used the pipeline can by splitted tasks that are executed by cpuTaskExecutor
+    // and waiting tasks. Waiting tasks can lock execution thread so they use separate threads from other executor.
+    constexpr const auto remoteDevice = false;
+
+    if (remoteDevice) {
+        m_pipeline = {
+            {cpuTaskExecutor,
+             [this] {
+                 OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin,
+                                    "TemplatePlugin::AsyncInferRequest::PreprocessingAndStartPipeline");
+                 get_template_infer_request()->infer_preprocess();
+                 get_template_infer_request()->start_pipeline();
```

Comment:
do we need to have method `get_template_infer_request()` ? Looks like we can use `inferRequest` here